packages/console/src/contexts/SubscriptionDataProvider/utils.ts

```diff
@@ -0,0 +1,52 @@
+import { pick } from '@silverhand/essentials';
+
+import { type NewSubscriptionCountBasedUsage } from '@/cloud/types/router';
+import { isCloud } from '@/consts/env';
+
+import { type SubscriptionUsageOptions } from './types';
+
+/* === For new pricing model === */
+const isSubscriptionUsageWithInLimit = <T extends keyof NewSubscriptionCountBasedUsage>(
+  { quotaKey, subscriptionUsage, subscriptionQuota, usage }: SubscriptionUsageOptions<T>,
+  inclusive = true
+) => {
+  // No limitations for OSS version
+  if (!isCloud) {
+    return true;
+  }
+
+  const usageValue = usage ?? subscriptionUsage[quotaKey];
+  const quotaValue = subscriptionQuota[quotaKey];
+
+  // Unlimited
+  if (quotaValue === null) {
+    return true;
+  }
+
+  if (typeof quotaValue === 'boolean') {
+    return quotaValue;
+  }
+
+  return inclusive ? usageValue <= quotaValue : usageValue < quotaValue;
+};
+
+export const hasSurpassedSubscriptionQuotaLimit = <T extends keyof NewSubscriptionCountBasedUsage>(
+  options: SubscriptionUsageOptions<T>
+) => {
+  console.log(
+    'hasSurpassedSubscriptionQuotaLimit',
+    pick(options, 'quotaKey', 'usage', 'subscriptionUsage', 'subscriptionQuota')
+  );
```

Comment:
Is this intended?

Reply:
Removed, this was for debugging purpose.